MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -4880,6 +4884,20 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 
 	startAutoGPUCapture(MVK_CONFIG_AUTO_GPU_CAPTURE_SCOPE_DEVICE, _physicalDevice->_mtlDevice);
 
+	if (getMVKConfig().autoGPUCaptureScope == MVK_CONFIG_AUTO_GPU_CAPTURE_SCOPE_ON_DEMAND) {
+		char pipeName[] = "/tmp/MoltenVKCapturePipe-XXXXXXXXXX";
+		mktemp(pipeName);
```

Comment:
`mktemp()` triggers a deprecation warning on build, and recommends using `mkstemp()`. Doing so might also simplify your code here. There is also `mkostemp()` to specify the file access modes, if needed.

If you absolutely need to use `mktemp()`, wrap this line, or this section, in

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
			    mktemp(pipeName);    // Deprecated but needed for interprocess transfers
#pragma clang diagnostic pop
```

See use elsewhere in MoltenVK.

Reply:
The problem is that he does in fact need to use `mktemp(3)` here, because he is not creating a regular file or a directory but a FIFO.

Which reminds me...

Reply:
The other `mk*temp` functions don't work, as far as I could tell there's no way to create a pipe using them. One alternative option is to create a temporary directory and then create a pipe with a static name in it. But to me that seems worse. I'll add the pragma thing.